### DISCUSSION TITLE
Using IP literal in ConnectionStart

### DISCUSSION THREAD

**Discussion by liwass** [microsoft/msquic#2587] [Q&A] [answered]
Documentation states that IP literal can be used for ServerName parameter, 
but in fact QUIC_STATUS_CONNECTION_REFUSED happens.
Should the server has to have certificate for IP for this to work?

**Comment by nibanks** [comment, marked as answer]
The protocol and MsQuic API allow it, but the server is still allowed to reject it, and the client might fail to accept the server's certificate if it doesn't have the IP in it. But going on the status code you shared, I'd say the server (app layer, above MsQuic) is rejecting your request. Server logs would likely be needed to verify definitively.
